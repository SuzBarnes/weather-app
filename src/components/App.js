import "../styles/App.css";
import React, { useState, useEffect } from "react";
import LocationDetails from "./LocationDetails";
import ForecastSummaries from "./ForecastSummaries";
import ForecastDetails from "./ForecastDetails";
import getForecast from "../requests/getForecast";
import SearchForm from "./SearchForm";

function App() {
  const [forecasts, setForecasts] = useState([]);
  const [location, setLocation] = useState({ city: "", country: "" });
  const [selectedDate, setSelectedDate] = useState(0);
  const [searchText, setSearchText] = useState("");
  const [errorMessage, setErrorMessage] = useState("");

  useEffect(() => {
    getForecast(setSelectedDate, setForecasts, setLocation);
  }, []);
  const selectedForecast = forecasts.find(
    (forecast) => forecast.date === selectedDate
  );
  const handleForecastSelect = (date) => {
    setSelectedDate(date);
  };
  const handleCitySearch = () => {
    setErrorMessage("");
    getForecast(
      setSelectedDate,
      setForecasts,
      setLocation,
      searchText,
      setErrorMessage
    );
  };

  return (
    <div className="weather-app">
      <LocationDetails
        city={location.city}
        country={location.country}
        errorMessage={errorMessage}
      />
      <SearchForm
        searchText={searchText}
        setSearchText={setSearchText}
        onSubmit={handleCitySearch}
      />
      <br />
      {!errorMessage && (
        <>
          <ForecastSummaries
            forecasts={forecasts}
            onForecastSelect={handleForecastSelect}
          />
          <br />
          {selectedForecast && <ForecastDetails forecast={selectedForecast} />}
        </>
      )}
    </div>
  );
}

export default App;
