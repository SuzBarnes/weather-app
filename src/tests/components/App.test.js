import { render } from "@testing-library/react";
import React from "react";
import App from "../../components/App";
import forecast from "../../components/ForecastSummaries";

describe("App", () => {
  it("renders correctly", () => {
    const { asFragment } = render(
      <App
        location={forecast.location}
        forecasts={forecast.forecasts}
        forecast={forecast[0]}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
