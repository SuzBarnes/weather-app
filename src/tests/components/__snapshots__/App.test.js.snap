// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<DocumentFragment>
  <div
    class="weather-app"
  >
    <h1>
      , 
    </h1>
    <div
      class="search-form"
    >
      <input
        type="text"
        value=""
      />
      <button
        class="searchformbutton"
        type="submit"
      >
        Search
      </button>
    </div>
    <br />
    <div
      class="forecast-summaries"
    />
    <br />
  </div>
</DocumentFragment>
`;
